File: 647. Palindromic Substrings/Approach03.py
from typing import *

class Solution:
    def __init__(self):
        self.result = 0

    def solve(self, s, i, j, n):
        while(i>=0 and j<n and s[i]==s[j]):
            self.result += 1
            i -= 1
            j += 1

    def countSubstrings(self, s: str) -> int:
        n = len(s)
        for i in range(n):
            self.solve(s, i, i, n)
            self.solve(s, i, i+1, n)
        return self.result

File: 647. Palindromic Substrings/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int result;
    void solve(string& s, int i, int j, int n){
        while( i>=0 && j<n && s[i]==s[j]){
            result++;
            i--;
            j++;
        }
    }
    int countSubstrings(string s) {
        int n = s.size();
        result = 0;

        for(int i=0; i<n; i++){
            solve(s, i, i, n);
            solve(s, i, i+1, n);
        }
        return result;
    }
};